openmetadata-ui/src/main/resources/ui/src/pages/UserPage/UserPage.component.tsx

```diff
@@ -113,27 +115,29 @@ const UserPage = () => {
       .finally(() => setIsLoading(false));
   };
 
-  const fetchEntities = (
+  const fetchEntities = async (
     fetchOwnedEntities = false,
     handleEntity: Dispatch<SetStateAction<AssetsDataType>>
   ) => {
     const entity = fetchOwnedEntities ? ownedEntities : followingEntities;
-    searchData(
-      fetchOwnedEntities
-        ? `owner.id:${userData.id}`
-        : `followers:${userData.id}`,
-      entity.currPage,
-      PAGE_SIZE,
-      ``,
-      '',
-      '',
-      myDataSearchIndex
-    )
-      .then((res) => {
-        const hits = res?.data?.hits?.hits as SearchEntityHits;
+    if (userData.id) {
+      try {
+        const response = await searchData(
+          fetchOwnedEntities
+            ? `owner.id:${userData.id}`
+            : `followers:${userData.id}`,
```

Comment:
as improvement lets extract this condition outside of try block and give fine query to the function